src/hub/ShareClassManager.sol

```diff
@@ -32,27 +33,36 @@ contract ShareClassManager is Auth, IShareClassManager {
     uint32 constant META_NAME_LENGTH = 128;
     uint32 constant META_SYMBOL_LENGTH = 32;
 
-    /// Storage
-    uint32 internal transient _epochIncrement;
     IHubRegistry public hubRegistry;
+
     mapping(bytes32 salt => bool) public salts;
-    mapping(PoolId poolId => uint32) public epochId;
+    mapping(ShareClassId scId => mapping(AssetId assetId => uint32)) public depositEpochId;
```

Comment:
`issueEpochId` and `depositEpochId` are both read in issue, and likewise for the revoke equivalents. I would add
```solidity
struct EpochIds {
  uint32 deposit;
  uint32 redeem;
  uint32 issue;
  uint32 revoke;
}
```

thus removing 1 SLOAD every time issue/revoke is called.

Reply:
This is exactly what we had, just with different naming. @mustermeiszer can we revert to that or it a major improvement for externals to have all epochs separately?

Reply:
As long as we move each counter in step-by-step logic, I am fine.

Reply:
Awesome! Then I will do this in a follow-up PR in order to get this merged.